plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -3562,6 +3576,297 @@ var _ = Describe("validator", func() {
+					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.SeedChange, true)()
+					shoot.Spec.SeedName = pointer.String(newSeed.Name + " other")
+
+					attrs := admission.NewAttributesRecord(&shoot, &oldShoot, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoot").WithVersion("version"), "binding", admission.Update, &metav1.UpdateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("Internal error occurred: could not find referenced seed"))
+				})
+			})
+
+			Context("shootIsBeingScheduled", func() {
+				It("should reject update of binding if target seed is marked for deletion", func() {
+					oldShoot.Spec.SeedName = nil
+					now := metav1.Now()
+					seed.DeletionTimestamp = &now
+					Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Update(&seed)).To(Succeed())
+
+					attrs := admission.NewAttributesRecord(&shoot, &oldShoot, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoot").WithVersion("version"), "binding", admission.Update, &metav1.UpdateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("cannot schedule shoot 'shoot' on seed 'seed' that is already marked for deletion"))
+				})
+			})
+
+			Context("shootIsBeingRescheduled a.k.a Control-Plane migration", func() {
+				BeforeEach(func() {
+					shoot.Spec.SeedName = pointer.String(newSeedName)
+				})
+
+				It("should reject update of binding if target seed is marked for deletion", func() {
+					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.SeedChange, true)()
+					now := metav1.Now()
+					newSeed.DeletionTimestamp = &now
+
+					attrs := admission.NewAttributesRecord(&shoot, &oldShoot, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoot").WithVersion("version"), "binding", admission.Update, &metav1.UpdateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("cannot schedule shoot '%s' on seed '%s' that is already marked for deletion", shoot.Name, newSeedName))
+				})
+
+				It("should reject update of binding, because target Seed doesn't have configuration for backup", func() {
+					defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.SeedChange, true)()
+
+					newSeed.Spec.Backup = nil
+
+					attrs := admission.NewAttributesRecord(&shoot, &oldShoot, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoot").WithVersion("version"), "binding", admission.Update, &metav1.UpdateOptions{}, false, nil)
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
```

Comment:
```suggestion
					Expect(err := admissionHandler.Admit(context.TODO(), attrs, nil)).To(HaveOccurred())
```

Same for expects below, so that we make sure that there was an error returned and it is not nil.